lib/licensed/sources/gradle.rb

```diff
@@ -4,132 +4,164 @@
 require "uri"
 require "json"
 require "net/http"
+require 'fileutils'
 
 module Licensed
   module Sources
     class Gradle < Source
 
-      DEFAULT_CONFIGURATIONS   = ["runtime", "runtimeClasspath"]
-      GRADLE_LICENSES_PATH     = ".gradle-licenses"
-      GRADLE_LICENSES_CSV_NAME = "licenses.csv"
+      DEFAULT_CONFIGURATIONS   = ["runtime", "runtimeClasspath"].freeze
+      GRADLE_LICENSES_PATH     = ".gradle-licenses".freeze
 
-      # Returns the configurations to include in license generation.
-      # Defaults to ["runtime", "runtimeClasspath"]
-      def configurations
-        @configurations ||= begin
-          if configurations = config.dig("gradle", "configurations")
-            Array(configurations)
-          else
-            DEFAULT_CONFIGURATIONS
+      class Dependency < Licensed::Dependency
+        GRADLE_LICENSES_CSV_NAME = "licenses.csv".freeze
+
+        class << self
+          # Returns a key to uniquely identify a name and version in the obtained CSV content
+          def csv_key(name:, version:)
+            "#{name}-#{version}"
+          end
+
+          # Loads and caches license report CSV data as a hash of :name-:version => :url pairs
+          #
+          # executable     - The gradle executable to run to generate the license report
+          # configurations - The gradle configurations to generate license report for
+          #
+          # Returns a hash of dependency identifiers to their license content URL
+          def load_csv(path, executable, configurations)
+            @csv ||= begin
+              gradle_licenses_dir = File.join(path, GRADLE_LICENSES_PATH)
+              Licensed::Sources::Gradle.gradle_command("generateLicenseReport", path: path, executable: executable, configurations: configurations)
```

Comment:
I removed `path.parent` here since it's much easier to reason about known path hierarchy requirements under the current directory than above and around it.  The dependency is passed `path: config.pwd` now instead of `gradle_licenses_dir` to account for this, with the `grade_licenses_dir` path being constructed (and deleted) in this method